src/app/model/ChessEngine.ts

```diff
@@ -17,6 +18,8 @@ export class ChessEngine implements ChessModel {
   readonly squaresWithPiece: SquareWithPiece;
   private promotingOnSquare: Square | undefined;
   private checkedKing: CheckedKing | undefined;
+  private castlingOnSquare: Square | undefined;
```

Comment:
Do utrzymania dobrego, modularnego kodu - powinnismy jak najbardiej ograniczac dostepnosc zmiennych. Nie tylko private / public itp. ale tez w zakresie. Jeśli ta zmienna jest potrzebna, między dwoma wywołaniami obiektu to wtedy dajemy ją do stanu obiektu. Ale im mniej stanu - tym lepiej. Czyli jeśli się da, to powinna być przekazywana tylko w zakresie funkcji, która tego potrzebuje.

Reply:
Czyli jak to zrobić, jeśli ta funkcja jest w innym miejscu?

Reply:
To jest teraz mega duży problem - tzn. niespodziewany 'side effect' metody.
Nie domyslilbym się, że `canMoveOnSquare` ustawia jakąś zmienną :) 
Patrząc po nazwie stawiam, że to mi po prostu mówi true/false. 
Więc albo:
- zmienic nazwe i typ
- ja proponuje po prostu nie ustawiac tej zmiennej tylko, zeby obie metody były niezalezne od drugiej.
Bo teraz wprowadzasz koniecznosc wywolania jednej i drugiej. 

A dodatkowo, widze, ze dalej nie obejrzeliscie functional programming, ktore bylo w materialach obowiazkowych do Node i pisze o tym od 2 miesiecy :) To nas uchroni przed takimi kwestiami. 

https://www.youtube.com/watch?v=uVK84yPnMMU&list=PLjHmWifVUNMLR1Oz3WYK7XRLEY5LvSOPu

Jeśli wprowadzamy koniecznosc wykonywania jednej metody po drugiej to powiem tak: nie jest dobrze :) Na pewno ktos kiedys uzyje tego w odwrotnej kolejnosci, czy zapomni wywolac jednej z nich.